pallets/committee/src/lib.rs

```diff
@@ -293,12 +312,9 @@ decl_module! {
         ///   - `M` is number of members,
         ///   - `P` is number of active proposals,
         ///   - `L` is the encoded length of `proposal` preimage.
-        #[weight = (T::DbWeight::get().reads_writes(6, 2) + 650_000_000, Operational, Pays::Yes)]
+        #[weight = <T as Trait<I>>::WeightInfo::close(Module::<T, I>::members().len() as u32)]
```

Comment:
We read `Members` in the extrinsic anyway. Subsequent reads of the same value should be cached by the DB.

Reply:
Yeah true, but it does render weight calculations into a function of storage, not just of the input, so not statically knowable anymore, which is what we've wanted to avoid as I understand it?

Reply:
Here the value of `Members` is treated as input to the weight computation. My idea is that obtaining it in `#[weight]` is offset against subsequent reads of that value inside the extrinsic. So we shouldn't be doing much extra work, while the computation does depend on that input and having it improves accuracy.

Reply:
All fair points. I guess it would be good to have a consistent policy though.

Reply:
I think we could _cache_ the number of members, to avoid loading the list of members (line 330). It will do that _dispatchable_ independent of the `Module::<T, I>::members()`. Instead of read a vector of members we will load just a u32.
Another optimization, that we could handle outside of this task is to cache the number of `nays` & `ayes`, in the same way that we do with `members()`. We could manage a _meta_ info like `voting_accountability`, which contains `number_of_nays` (u32) and `number_of_ayes` (u32).

Reply:
I fixed the maximum number of committee members to 1000 and dropped the length argument.